elliottlib/find_cve_flaws.py

```diff
@@ -0,0 +1,90 @@
+import requests
+import ssl
+from elliottlib import constants, bzutil
+from requests_kerberos import HTTPKerberosAuth
+
+
+def get_attached_tracker_bugs(bzapi, advisory_id):
+    return bzapi.getbugs([
+        bug['id']
+        for bug in get_all_attached_bugs(advisory_id)
+        if is_tracker_bug(bug)
+    ])
+
+
+def get_all_attached_bugs(advisory_id):
+    return [bug['bug'] for bug in get_advisory(advisory_id)['bugs']['bugs']]
+
+
+def is_tracker_bug(bug):
+    return 'Security' in bug['keywords'] and 'SecurityTracking' in bug['keywords']
+
+
+def is_flaw_bug(bug):
+    return bug.product == 'Security Response'
+
+
+def get_advisory(advisory_id):
+    res = requests.get(
+        '{}/api/v1/erratum/{}'.format(constants.errata_url, advisory_id),
+        verify=ssl.get_default_verify_paths().openssl_cafile,
+        auth=HTTPKerberosAuth(),
+    )
+    return res.json()
+
+
+def get_corresponding_flaw_bugs(bzapi, tracker_bugs):
+    blocking_bugs = bzapi.getbugs(unique(flatten([t.blocks for t in tracker_bugs])))
+    return [flaw_bug for flaw_bug in blocking_bugs if is_flaw_bug(flaw_bug)]
+
+
+def is_first_fix(bzapi, flaw_bug, current_target_release, tracker_ids_to_be_ignored=[]):
+    other_flaw_trackers = bzapi.getbugs([
+        t for t in flaw_bug.depends_on
+        if t not in tracker_ids_to_be_ignored
+    ])
+
+    def _filter_tracker(bug):
+        if all([
+            bug.product == 'OpenShift Container Platform',
+            bug.target_release[0] <= current_target_release,
```

Comment:
huh. two things about this.

1. i don't remember it mattering if the target release is lower. however we _do_ want to exclude target releases for a different major version (3 vs 4).
1. string comparison will break when we get to 4.10... transform with something like [this](https://github.com/openshift/elliott/pull/163/files#diff-65f7a43ce00364aa8475ae3b69500e431fbf5892cf10648158d8fad6428c649dR263) if comparison needed

Reply:
:+1: